docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+    REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+    SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+    SomeEnvironment/Prod:2    Active       5         5         5            5        
+    ```
+
+Rollback to an earlier revision
+-------------------------------
+
+Let's say that revision 2 is actually broken, and we need to roll back
+to revision 1:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+SomeEnvironment/Prod:2    Active       5         5         5            5        
+```
+
+**Option 1**: Allow specifying earlier revisions to deploy. That
+revision becomes active again, and we deactivate the other revisions.
+
+``` {.shell}
+$ aws ecs deploy-environment --environment-name SomeEnvironment/Prod --environment-revision 1
+{ "DeploymentId": "SomeEnvironment/Prod:1@2017090611472301" }
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying    0         5         5            5        
+SomeEnvironment/Prod:2    Undeploying  6         0         0            0        
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Active       5         5         5            5        
+SomeEnvironment/Prod:2    Reverted     0         0         0            0        
+```
+
+**Option 2**: Rollback copies the earlier revision to a new revision,
+and deploys that. The old revision remains inactive, and is not reused.
+
+``` {.shell}
+$ aws ecs deploy-environment --environment-name SomeEnvironment/Prod --environment-revision 1
+{ "DeploymentId": "SomeEnvironment/Prod:3@2017090611472301" }
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+SomeEnvironment/Prod:2    Undeploying  6         6         6            6        
+SomeEnvironment/Prod:3    Deploying    0         0         0            0        
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+SomeEnvironment/Prod:2    Inactive     0         0         0            0        
+SomeEnvironment/Prod:3    Active       6         6         6            6        
```

Comment:
I like these status displays, but environments are going to accumulate a lot of deployment revisions over time.  Do we have thoughts on how far back we keep/display history?  May want `--limit N` arg to let user tell us how much they want to see.

Reply:
Indeed, the idea is to hide `Inactive` by default, and provide a `showInactive=true` option and pagination APIs; will add a bit at the top to that effect.